pipeline-scripts/buildlib.groovy

```diff
@@ -1564,6 +1564,39 @@ def get_owners(doozerOpts, images, rpms=[]) {
     return yamlData
 }
 
+def setBuildType() {
+    if (!isTriggeredByOrganicLifeForms()) {
+        echo "Assuming *PROD* build, triggered by automation"
+        scratch = false
```

Comment:
I am concerned about the use of global variable `scratch` here. People have to remember to define this global variable when calling from a Jenkinsfile. I would suggest returning this value instead.

Also the function name `setBuildType` sounds confusing. Would prefer a name more specific like `getBuildPurpose` `shouldSkipTag` or something.